libraries/ae-inventory-processor/src/main/java/org/metaeffekt/core/inventory/processor/patterns/ComponentPatternProducer.java

```diff
@@ -339,7 +340,9 @@ private void markFilesCoveredByComponentPatterns(List<MatchResult> matchedCompon
                     if (LOG.isDebugEnabled()) {
                         LOG.debug("No files matched for component pattern {}.", matchResult.componentPatternData.createCompareStringRepresentation());
                     }
-                    matchResultsWithoutFileMatches.add(matchResult);
+                    if (cpd.get(COMPONENT_SOURCE_TYPE) == null || !cpd.get(COMPONENT_SOURCE_TYPE).equals("rpm") || !cpd.get(COMPONENT_SOURCE_TYPE).equals("alpm")) {
```

Comment:
Contributors must indicate that by adding an additional attribute to the contributions patterns. This condition then checks the attribute rather than the source type.